src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,126 @@
 package core.basesyntax;
 
+import java.util.Collection;
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int length;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        length = 0;
+    }
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity >= 0) {
+            elementData = (T[]) new Object[initialCapacity];
+        } else {
+            throw new IllegalArgumentException("Illegal Capacity: " + initialCapacity);
+        }
+        length = 0;
+    }
+
+    public ArrayList(Collection<? extends T> c) {
+        elementData = (T[]) c.toArray();
+    }
 
     @Override
     public void add(T value) {
-
+        increaseArray();
+        elementData[length] = value;
+        length++;
     }
 
     @Override
     public void add(T value, int index) {
+        increaseArray();
+        if (indexIsExist(index)) {
+            length++;
+            System.arraycopy(elementData, index, elementData, index + 1, length - index - 1);
+            elementData[index] = value;
 
+        } else {
```

Comment:
this else block is redundant, you can optimize it like this 
if (!indexIsExist(index)) {
   throw new Ar...
}

Reply:
Done